SpatialGDK/Source/SpatialGDK/Private/Interop/Connection/SpatialEventTracer.cpp

```diff
@@ -0,0 +1,226 @@
+
+		// convert the property to a FJsonValue
+		if (UnrealStrProperty *StringProperty = CastField<UnrealStrProperty>(Property))
+		{
+			AddTraceEventStringField(VariableName, StringProperty->GetPropertyValue(Value));
+		}
+		else if (UnrealObjectProperty* ObjectProperty = CastField<UnrealObjectProperty>(Property))
+		{
+			UObject* Object = ObjectProperty->GetPropertyValue(Value);
+			if (Object)
+			{
+				AddTraceEventStringField(VariableName, Object->GetName());
+				if (AActor* Actor = Cast<AActor>(Object))
+				{
+					FString KeyString = VariableName + TEXT("Position");
+					FString ValueString = Actor->GetTransform().GetTranslation().ToString();
+					AddTraceEventStringField(KeyString, ValueString);
+				}
+			}
+			else
+			{
+				AddTraceEventStringField(VariableName, "Null");
+			}
+		}
+		else // Default
+		{
+			FString StringValue;
+			Property->ExportTextItem(StringValue, Value, NULL, NULL, PPF_None);
+			AddTraceEventStringField(VariableName, StringValue);
+		}
+
+		// else if (UEnumProperty* EnumProperty = Cast<UEnumProperty>(Property))
+		// else if (UNumericProperty *NumericProperty = Cast<UNumericProperty>(Property))
+		// else if (UBoolProperty *BoolProperty = Cast<UBoolProperty>(Property))
+		// else if (UTextProperty *TextProperty = Cast<UTextProperty>(Property))
+		// else if (UArrayProperty *ArrayProperty = Cast<UArrayProperty>(Property))
+		// else if (USetProperty* SetProperty = Cast<USetProperty>(Property))
+		// else if (UMapProperty* MapProperty = Cast<UMapProperty>(Property))
+		// else if (UStructProperty *StructProperty = Cast<UStructProperty>(Property))
+
+	}
+
+	// TODO(EventTracer): implement and call AddTargetObjectInfoToEventData
+	// TODO(EventTracer): implement and call AddComponentIdInfoToEventData
+
+	TraceEvent.data = EventData;
+	Trace_EventTracer_AddEvent(EventTracer, &TraceEvent);
+	Trace_EventData_Destroy(EventData);
+
+	return CurrentSpanId;
+}
+
+void SpatialEventTracer::Enable(const FString& FileName)
+{
+	Trace_EventTracer_Parameters parameters = {};
+	parameters.user_data = this;
+	parameters.callback = &SpatialEventTracer::TraceCallback;
+	EventTracer = Trace_EventTracer_Create(&parameters);
+	Trace_EventTracer_Enable(EventTracer);
+
+	bEnabled = true;
+
+	UE_LOG(LogSpatialEventTracer, Log, TEXT("Spatial event tracing enabled."));
+
+	// Open a local file
+	const FString FolderPath = FPaths::Combine(FPaths::ProjectSavedDir(), TEXT("EventTracing"));
+	const FString FullFileName = FString::Printf(TEXT("EventTrace_%s_%s.trace"), *FileName, *FDateTime::Now().ToString());
+	const FString FilePath = FPaths::Combine(FolderPath, FullFileName);
+
+	IPlatformFile& PlatformFile = FPlatformFileManager::Get().GetPlatformFile();
+	if (PlatformFile.CreateDirectoryTree(*FolderPath))
+	{
+		UE_LOG(LogSpatialEventTracer, Log, TEXT("Capturing trace to %s."), *FilePath);
+		Stream = Io_CreateFileStream(TCHAR_TO_ANSI(*FilePath), Io_OpenMode::IO_OPEN_MODE_WRITE);
```

Comment:
[nit] You could make `Stream` a unique ptr type with custom deleter which calls `Io_Stream_Destroy` to avoid having to explicitly destroy it in `Disable`.